include/mbgl/mtl/texture2d.hpp

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <mbgl/gfx/texture2d.hpp>
+#include <mbgl/mtl/mtl_fwd.hpp>
+#include <mbgl/util/image.hpp>
+
+#include <Foundation/NSSharedPtr.hpp>
+#include <Metal/MTLPixelFormat.hpp>
+
+#include <memory>
+
+namespace mbgl {
+namespace mtl {
+
+class Context;
+class RenderPass;
+
+class Texture2D : public gfx::Texture2D {
+public:
+    Texture2D(Context& context_);
+    ~Texture2D() override;
+
+public:
```

Comment:
The `public` keyword is used three times in a row in this class, and `private` twice in a row.

Reply:
I believe @alexcristici's render target work makes major changes to `Texture2D`.